### DISCUSSION TITLE
Filling the region between two curves in Cartesian math plot

### DISCUSSION THREAD

**Discussion by psads-git** [cetz-package/cetz#418] [Q&A] [answered]
I am using the code below, which plots the following graph:

![image](https://github.com/johannes-wolf/cetz/assets/75945439/5ad7f75c-cd80-4912-a6ab-6cef3ea3b41b)

```
#import "@preview/cetz:0.1.2"

#cetz.canvas({
  import cetz.plot
  import cetz.palette
  
  plot.plot(size: (4,4), x-tick-step: 1, axis-style: "left", {
  plot.add(domain: (-4, 4),
  x => (4-calc.pow(x, 2)),
  fill: true, style: palette.tango)
  plot.add(domain: (-4, 4),
  x => (4-2*x),
  fill: true, style: palette.tango)})
})

```
There is some aspects that I want to improve:

- Constrain the filling _only_ to the region between the parabola and the straight line.

- At the two intersession points, a small black-filled ball should be placed.

- The $y$ axis should cross the $x$ axis at $x=0$.

- The $x$ axis should cross the $y$ axis at $y=0$.

- The frequency of the axes ticks labels should be much lower, so that the numbers appear more spaced in between.

Please, be patient with a newbie CeTZ user.

**Comment by johannes-wolf** [comment, marked as answer]
You can use the `0.2.0` branch (or wait for the 0.2.0 release), which includes a `plot.add-fill-between` function which does the filling you want. For the intersections, you have to place them yourself right now, by adding another plot with no lines (`stroke: none`), but marks: `mark: "o"` and your intersection points as data (list of points).

- Cross axes at (0,0): `axis-style: "school-book"` should do what you want.
- Tick frequency: `x-tick-step: <your-delta>`. If you want additional tick marks without labels, you can enable minor ticks, by setting `x-minor-tick-step: ..` to some value.

Example:
```typst
#cetz.canvas({
  import cetz.plot
  import cetz.palette

  plot.plot(size: (4,4),
    x-tick-step: 2,
    x-minor-tick-step: 1,
    y-tick-step: 5,
    y-minor-tick-step: 1,
    axis-style: "school-book", {
    plot.add-fill-between(domain: (-4, 4),
      x => (4-calc.pow(x, 2)),
      x => (4-2*x),
      style: palette.tango)
    plot.add(((0,4), (2,0)),
      mark: "o",
      mark-style: (stroke: none, fill: black),
      style: (stroke: none))
  })
})
```
![image](https://github.com/johannes-wolf/cetz/assets/519002/abc77fb4-4954-4dc5-9a22-a89e51346512)

**Reply by johannes-wolf** [reply]
I guess I should add a style option to set the `0` tick offset separately… #419.

**Reply by psads-git** [reply]
Thanks, @johannes-wolf: That is what I am needing! Unfortunately, I am getting this error:^

```
error: failed to decompress package (failed to iterate over archive)
    ┌─ /home/x/x.typ:131:8
    │
131 │ #import "@preview/cetz:0.2.0"
    │         ^^^^^^^^^^^^^^^^^^^^^
```

**Reply by johannes-wolf** [reply]
Yes, 0.2.0 is not yet available/released.

**Reply by psads-git** [reply]
By using version 0.2.0 locally, it works fine. Thank you very much, @johannes-wolf!
